iroh/tests/sync.rs

```diff
@@ -30,38 +38,55 @@ fn test_runtime() -> runtime::Handle {
 fn test_node(
     rt: runtime::Handle,
     addr: SocketAddr,
+    secret_key: SecretKey,
 ) -> Builder<iroh_bytes::store::mem::Store, store::memory::Store, DummyServerEndpoint> {
     let db = iroh_bytes::store::mem::Store::new(rt.clone());
     let store = iroh_sync::store::memory::Store::default();
-    Node::builder(db, store).runtime(&rt).bind_addr(addr)
+    Node::builder(db, store)
+        .secret_key(secret_key)
+        .derp_mode(DerpMode::Disabled)
+        .runtime(&rt)
+        .bind_addr(addr)
 }
 
-async fn spawn_node(
+fn spawn_node(
```

Comment:
why?

Reply:
The function is not `async fn` so that we can take a `&mut` borrow on the `rng` without capturing that `&mut` lifetime in the returned future. This allows to call it in a loop while still collecting the futures before awaiting them alltogether (see [`spawn_nodes`])

Will add the explanation as a comment.